airflow/triggers/base.py

```diff
@@ -115,6 +116,19 @@ async def cleanup(self) -> None:
         and handle it appropriately (in async-compatible way).
         """
 
+    def should_cleanup(self, termination_reason: TriggerTerminationReason | None) -> bool:
+        """
+        Check the trigger should be cleaned up or not base on the context.
+
+        :param termination_reason: The reason for terminating the trigger.
+        Since the trigger could be terminated for various reasons, like triggerer restart or reassigned to
+        another triggerer, this method allows the trigger to decide whether it should be cleaned up under the
+        current circumstances.
+
+        By default, will always return True, override this method base custom requirements.
```

Comment:
Is this only for testing purposes? The docstring should say so instead if that’s the case.

Reply:
Some text should also be added to the documentation (probably in the *Deferrable Operators & Triggers* page?) to teach users when/how this can be used.

Reply:
Yes, let me see how to add it

Reply:
Add one section in the Deferrable Operators & Triggers page, please help to review @uranusjr